### PR TITLE
=act Experiment with LineNumbers for Lambda's #24480

### PR DESCRIPTION
**Pull request by jlprat**
Refs: #24480
All credit to @rkuhn whose code I just uncommented

Feedback welcome! With such examples it works for both Scala 2.12 and 2.11.12

### CODE REVIEW

File: akka-actor/src/main/scala/akka/util/LineNumbers.scala [hunk 1]
```diff
@@ -196,10 +197,10 @@ object LineNumbers {
       val writeReplace = c.getDeclaredMethod("writeReplace")
       writeReplace.setAccessible(true)
       writeReplace.invoke(l) match {
-        //        case serialized: SerializedLambda ⇒
-        //          if (debug) println(s"LNB:     found Lambda implemented in ${serialized.getImplClass}:${serialized.getImplMethodName}")
-        //          Option(c.getClassLoader.getResourceAsStream(serialized.getImplClass + ".class"))
-        //            .map(_ -> Some(serialized.getImplMethodName))
+        case serialized: SerializedLambda ⇒
+          if (debug) println(s"LNB:     found Lambda implemented in ${serialized.getImplClass}:${serialized.getImplMethodName}")
+          Option(c.getClassLoader.getResourceAsStream(serialized.getImplClass + ".class"))
```
Comment (patriknw) [inline akka-actor/src/main/scala/akka/util/LineNumbers.scala]: i double-checked that this InputStream is closed

**Review by patriknw — APPROVED** [review]
great that it works, do we see any risks/drawbacks of enabling this?

**Review by johanandren — APPROVED** [review]

**Review by ktoso — APPROVED** [review]
Naisu, let's take it :)

**Comment by SethTisue** [comment]
this is now consistently failing in the Scala 2.12 community build at e.g. https://scala-ci.typesafe.com/view/scala-2.12.x/job/scala-2.12.x-integrate-community-build/2625/

```
[akka-more] [info] LineNumbers
[akka-more] [info]   when writing Scala
[akka-more] [info]   - must work for small functions *** FAILED *** (25 milliseconds)
[akka-more] [info]     LineNumberSpecCodeForScala.scala:12 did not equal NoSourceInfo (LineNumberSpec.scala:25)
```

it also failed for me locally, outside of dbuild, when I just did `++2.12.4` followed by `akka-actor-tests/testOnly *LineNumberSpec`

I'm a bit puzzled that this passed CI

**Comment by jlprat** [comment]
CI runs on 2.11.
When submitted the PR I tried with 2.12.3 and tests were green.

**Comment by rkuhn** [comment]
I suppose this means that Scala in the community build now has more source info than 2.12.3, which is a good thing!

**Comment by jlprat** [comment]
I will work on fixing this today :)

**Comment by SethTisue** [comment]
any luck?

**Comment by SethTisue** [comment]
looks like `LineNumbers` got fixed, one way or the other... green at https://scala-ci.typesafe.com/view/scala-2.12.x/job/scala-2.12.x-integrate-community-build/2848/

**Pull request merged**